awx/api/serializers.py

```diff
@@ -128,7 +128,7 @@
     'inventory_source': ('source', 'last_updated', 'status'),
     'custom_inventory_script': DEFAULT_SUMMARY_FIELDS,
     'source_script': ('name', 'description'),
-    'role': ('id', 'role_field'),
+    'role': ('id', 'role_field', 'content_type'),
```

Comment:
This could have a fairly significant effect on the number of queries needed to generate list views. However, I don't actually know what views make use of the default role summary fields.

Reply:
Good point, though Django does cache the full set of `ContentType` objects, so there shouldn't actually be any db hits.

Reply:
I thought that the cache only worked if you used the default content type manager, and using it as a ForeignKey or in a GenericForeignKey like here, it'll still go fetch it for each. I agree that having the `content_type_id` field from a role object would allow it to be looked up straight from the cache, but I don't think it'll do that unless you did something like `ContentType.objects. get_for_id(role.content_type_id)`

Reply:
but again, for this to ever affect anything at all, some model would have to have a field named `role`, and I don't think this is particularly common.

Reply:
Yeah, I (after a bit of fumbling) tested the queries involved and it _does_ hit the db.

```
In [34]: context.initial_queries, context.final_queries
Out[34]: (42, 2980)
```

```
In [33]: for x in context.captured_queries:
    ...:     if 'content_type' in x['sql']:
    ...:         print(x['sql'])
(...)
SELECT "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" WHERE "django_content_type"."id" = 20
```

Reply:
Even if I rearrange things, it's going to be doing an N+1 with regards to the Role objects, so this is a dead end.  I'm going to rethink this a bit.